lib/btc/BtcRpc.js

```diff
@@ -53,65 +64,81 @@ class BtcRpc {
     if (passphrase === undefined) {
       passphrase = await passwordPromptAsync('> ');
     }
-    await this.walletUnlock({ passphrase, time: 10 });
+    await this.walletUnlock({ passphrase, time: 10800 });
     const tx = await this.sendToAddress({ address, amount });
     await this.walletLock();
     return tx;
   }
 
-  async unlockAndSendToAddressMany({ payToArray, passphrase, time = 10800 }) {
-    function RpcException(message) {
-      const e = new Error(message);
-      e.name = 'failedRequest';
-      e.data = errorObject;
-      return e;
-    }
-
+  async unlockAndSendToAddressMany({ payToArray, passphrase, time=10800, maxValue=10*1e8, maxOutputs=1 }) {
     if (passphrase === undefined) {
       passphrase = await passwordPromptAsync('> ');
     }
+    if (payToArray.length < maxOutputs) {
+      maxOutputs = payToArray.length;
+    }
     await this.walletUnlock({ passphrase, time });
-    this.emitter.emit('unlocked', time);
-    let someRequestFailed;
-    const errorObject = { success: {}, failure: {} };
-    const result = [];
-    for (const [i, request] of payToArray.entries()) {
-      const { address, amount } = request;
-      const emitData = {
-        address: request.address,
-        amount: request.amount,
-        id: request.id
+    let payToArrayResult = [];
+    while(payToArray.length) {
+      let currentValue = 0;
+      let currentOutputs = 0;
+      let batch = {};
+      let tempArray = [];
+      while(currentValue < maxValue && currentOutputs < maxOutputs) {
+        let tx = payToArray.shift();
+        tempArray.push(tx);
+        let address = tx.address;
+        let amount = tx.amount;
+        if (address in batch) {
+          let currentAmount = batch[address];
+          amount = currentAmount + amount;
+        }
+        if (payToArray.length < maxOutputs) {
+          maxOutputs = payToArray.length;
+        }
+        batch[address] = amount;
+        currentValue += amount;
+        currentOutputs++;
+      }
+      let emitData = {
+        data: {},
+        txid: '',
+        error: null
       };
-      this.emitter.emit('attempt', emitData);
       try {
-        const txid = await this.sendToAddress({ address, amount });
-        errorObject.success[i] = txid;
+        let txid = await this.sendMany({ batch });
+        emitData.data = batch;
         emitData.txid = txid;
-        this.emitter.emit('success', emitData);
-        result.push(txid);
-      } catch (error) {
-        errorObject.failure[i] = error;
+        let voutCounter = 1;
+        for (let tx of tempArray) {
```

Comment:
I don't really understand this block decorating tx with txid and vout, where does that get used?

Reply:
it gets used on the settlement side, `tempArray` is basically each settlement in a single batch (and it preserves duplicates unlike the batch) and I am just adding the `txid` and `vout` that the settlement is a part of and concating that to a main array which I return back. The main array will have a 1 to 1 correlation to the `payToArray` passed in, just with the `txid` and `vout` also attached